src/definitions/modules/popup.js

```diff
@@ -510,9 +510,14 @@ $.fn.popup = function(parameters) {
               $popupOffsetParent = module.get.offsetParent($popup),
               targetElement      = $target[0],
               isWindow           = ($boundary[0] == window),
-              targetPosition     = (settings.inline || (settings.popup && settings.movePopup))
+              usePosition        = settings.inline || (settings.popup && settings.movePopup),
+              targetOffset       = $target.offset(),
+              targetPosition     = usePosition
                 ? $target.position()
-                : $target.offset(),
+                : targetOffset,
```

Comment:
`$target.position()` will never be used anymore later on, even if `usePosition` is true . The later code always refers to targetOffset later.
So we may use either targetPosition or targetOffset

I suggest to reduce it to either
```javascript
// remove the extra targetOffset declaration
//targetOffset       = $target.offset(),
targetPosition = $target.offset();
....

top     : usePosition ? Math.max(targetPosition.top - offsetParentOffset.top, 0) : targetPosition.top,
left    : usePosition ? Math.max(targetPosition.left - offsetParentOffset.left, 0) : targetPosition.left,

```

or

```javascript
// remove targetPosition at all and always work with targetOffset
targetOffset       = $target.offset(),
//targetPosition = $target.offset();
....

top     : usePosition ? Math.max(targetOffset.top - offsetParentOffset.top, 0) : targetOffset.top,
left    : usePosition ? Math.max(targetOffset.left - offsetParentOffset.left, 0) : targetOffset.left,

```

Reply:
Thanks !!
I chosed the 2nd suggestion.
Because `targetPosion = $target.offset()` seems weird: it stores `offset` instead of `position`...

Reply:
I also simplified (and optimized) code, in a same fashion for other variables.